### ISSUE TITLE
Normalization methods should return a new DGEList instance

### ISSUE THREAD

**Issue by apfejes** [r-bioinformatics/edgePy#61]
Convert the normalization methods to return a new DGEList instance, instead of modifying the exiting one.

**Comment by sdhutchins** [comment]
Anyone address this yet @apfejes ?

**Comment by apfejes** [comment]
Nope - no one has touched it yet. Supper easy one to take on.

**Comment by sdhutchins** [comment]
Yeah, I figure I might as well jump in and do some basic tasks to help push this towards completion.

**Comment by apfejes** [comment]
That's awesome!  Let us know how we can support you!

**Comment by Carbyne** [comment]
I'm wondering how we should make this work if other functions (ex. RPKM) are dependent on CPM. That would mean creating a new DGEList two times, which doesn't sound very efficient. It might help if there was a way to create a new DGEList without re-validating the data that's already been validated once.

**Comment by apfejes** [comment]
I think there's a bit of confusion.

The purpose of this issue is to return a new DGE object, but there's nothing to stop the person using the object from writing:

` dge_item = dge_item.rpkm()`

which replaces the existing object, and gives you a new one in which the self.counts feature now consists of RPKM values instead of raw counts or CPM. 

the .rpkm() and .cpm() functions simply transform what's in the self.counts attribute - or after this issue is completed, return a copy of the DGEList with the self.counts in whichever units were called.

**Comment by Carbyne** [comment]
Ah I see, then there shouldn't be a problem as long as we simply create a copy of the DGEList instead of initializing a new one :thumbsup:

**Comment by apfejes** [comment]
Should be fixed on branch af_rpm.

**Comment by apfejes** [comment]
Can be closed once branch #75  is merged.
